Library/ENV/scm/git

```diff
@@ -23,6 +23,10 @@ def exec(*args)
   # prevent fork-bombs
   arg0 = args.first
   return if arg0 =~ /^#{F}/i || Pathname.new(arg0).realpath == SELF_REAL
+  if ARGV == %w[--homebrew=print-path]
```

Comment:
You have  convinced me `--homebrew=<feature>`. It's a better idea. 

Also may I propose to handl arguments this way:
```Ruby
ARGV.dup.each_with_index do |arg, i|
  if arg.start_with? "--homebrew="
    ARGV.delete(i)
    # parse feature from arg
  end
end
```

The reason is I'm thinking to use `ENV/scm/git --homebrew=ignore-old-vendor-version --version` for vendoring Git. The idea is to control git resolution logic and execute git command at the same time.